compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -316,12 +316,14 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 ),
                 ExprKind::Struct(se) => {
                     let rest = match &se.rest {
-                        StructRest::Base(e) => Some(self.lower_expr(e)),
+                        StructRest::Base(e) => hir::Rest::Base(self.lower_expr(e)),
                         StructRest::Rest(sp) => {
-                            let guar = self.dcx().emit_err(BaseExpressionDoubleDot { span: *sp });
-                            Some(&*self.arena.alloc(self.expr_err(*sp, guar)))
+                            if !self.tcx.features().default_field_values() {
+                                self.dcx().emit_err(BaseExpressionDoubleDot { span: *sp });
```

Comment:
This error should include the same information as the feature gate in nightly, telling people to enable the `feature` if they want to use it *if* the underlying struct *has* default fields.